.github/workflows/build.yml

```diff
@@ -39,43 +45,49 @@ jobs:
               run: |
                   curl -s -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" -i https://api.github.com/users/octocat | grep x-ratelimit
 
-            - name: Run Linters and Test (Linux and macOS)
-              if: runner.os != 'Windows'
-              shell: bash
-              run: ./gradlew check
-
-            # Termination of AppMap CLI processes is failing with IC-2021.3 for unknown reasons (process.destroy() etc.)
-            - name: Run Linters and Test with 2023.2 (Windows)
-              if: runner.os == 'Windows'
+            - name: Run Linters and Test
               shell: bash
-              run: ./gradlew check -PideVersion=IC-2023.2.1
-
-            - name: Verify plugin
-              shell: bash
-              run: ./gradlew verifyPlugin
-
-            - name: Run Plugin Verifier
-              shell: bash
-              # JetBrains plugin verifier should always report the same results, regardless of the OS
-              if: runner.os == 'Linux'
-              run: ./gradlew runPluginVerifier
+              run: ./gradlew -PideVersion=${{ matrix.ide_version }} check
 
             - name: Upload Test Reports
-              uses: actions/upload-artifact@v3
+              uses: actions/upload-artifact@v4
               if: always()
               with:
-                  name: test-reports-${{ matrix.os }}
+                  name: test-reports-${{ matrix.os }}-${{ matrix.ide_version }}
                   path: |
                       build/reports/
                       **/build/reports/
 
             - name: Save AppMaps
-              uses: actions/cache/save@v3
-              if: runner.os == 'Linux'
+              uses: actions/cache/save@v4
+              if: runner.os == 'Linux' && matrix.ide_version == 'IC-2023.1'
               with:
                   path: ./tmp/appmap
                   key: appmaps-${{ github.sha }}-${{ github.run_attempt }}
 
+    verify:
+        name: Verify Plugin
+        runs-on: ubuntu-latest
+        needs: [test]
+        timeout-minutes: 45
+        steps:
+            - name: Fetch Sources
+              uses: actions/checkout@v4
+
+            - name: Setup Java
+              uses: actions/setup-java@v4
+              with:
+                  distribution: temurin
+                  java-version: ${{ env.JAVA_VERSION }}
+
+            - name: Verify Plugin Structure
+              shell: bash
+              run: ./gradlew verifyPlugin
```

Comment:
Only somewhat related, but this is failing on main
https://github.com/getappmap/appmap-intellij-plugin/actions/runs/9914172346/job/27392737457#step:8:1

Reply:
https://github.com/getappmap/appmap-intellij-plugin/pull/742 should fix the CI failure

Reply:
CI of this PR is failing atm because the current CI expired (I think). I'll rebase when #743 is merged.